presto-hive/src/main/java/com/facebook/presto/hive/util/InternalHiveSplitFactory.java

```diff
@@ -163,14 +166,23 @@ public String getPartitionName()
                 // skip zero-width block, except in the special circumstance: slice is empty, and the block covers the empty slice interval.
                 continue;
             }
-            blockBuilder.add(new InternalHiveBlock(blockEnd, getHostAddresses(blockLocation)));
+
+            List<HostAddress> addresses = getHostAddresses(blockLocation);
+            if (!schedulerUsesAddresses && !(forceLocalScheduling && hasRealAddress(addresses))) {
```

Comment:
nit:

```
List<HostAddress> addresses = ImmutableList.of();
if(schedulerUsesAddresses || (forceLocalScheduling && hasRealAddress(addresses))){
    addresses = getHostAddresses(blockLocation);
}
```

Reply:
do you mean:
```
List<HostAddress> addresses = ImmutableList.of();
List<HostAddress> hostAddresses = getHostAddresses(blockLocation);
if(schedulerUsesAddresses || (forceLocalScheduling && hasRealAddress(hostAddresses))){
    addresses = hostAddresses;
}
```

Reply:
Oh, nevermind. I didn't realize `hasRealAddress` accepts `HostAddress`